tennis-scheduler-web-app/src/components/EditProfile.jsx

```diff
@@ -0,0 +1,103 @@
+import React, { useState, useEffect } from "react"
+import { Link, useParams } from "react-router-dom";
+import '../styles/ProfileComponent.css';
+
+import { getLoggedUser } from "../api/PersonApi";
+
+export const EditProfile = () => {
+    const [user, setUser] = useState({ id: 0, firstName: "", lastName: "", email: "", phoneNumber: "", birthday: "", gender: "", address: { id: 0, street: "", city: "", number: 0, country: "" } });
+    const [formErrors, setFormErrors] = useState({});
+
+    useEffect(() => {
+        getLoggedUser().then((response) => setUser(response.data)).catch((error) => console.log(error))
+    }, []);
+
+    const handleChange = e => {
+        const { name, value } = e.target;
+        setUser((user) => ({ ...user, [name]: value }));
+    };
+
+    const editProfileSubmit = e => {
+        e.preventDefault();
+        setFormErrors(validation(user));
+    };
+
+    const validation = form => {
+        const errors = {};
+        if (!form.firstName) errors.name = "Name is required!";
+        if (!form.lastName) errors.surname = "Surname is required!";
```

Comment:
Why only validation for these fields?

Reply:
I think there should be validation even if fields are not necessary,  if user decide to add phone number or to change it you should check that